rad/src/error.rs

```diff
@@ -253,11 +277,223 @@ pub enum RadError {
         achieved, required
     )]
     InsufficientConsensus { achieved: f64, required: f64 },
+    /// The request contains too many sources.
+    #[fail(display = "The request contains too many sources")]
+    RequestTooManySources,
+    /// The script contains too many calls.
+    #[fail(display = "The script contains too many calls")]
+    ScriptTooManyCalls,
+    /// At least one of the source scripts is not a valid CBOR-encoded value.
+    #[fail(display = "At least one of the source scripts is not a valid CBOR-encoded value")]
+    SourceScriptNotCBOR,
+    /// The CBOR value decoded from a source script is not an Array.
+    #[fail(display = "The CBOR value decoded from a source script is not an Array")]
+    SourceScriptNotArray,
+    /// The Array value decoded form a source script is not a valid RADON script.
+    #[fail(display = "The Array value decoded form a source script is not a valid RADON script")]
+    SourceScriptNotRADON,
+    /// Math operator caused an underflow.
+    #[fail(display = "Math operator caused an underflow")]
+    Underflow,
+    /// Tried to divide by zero.
+    #[fail(display = "Tried to divide by zero")]
+    DivisionByZero,
+    /// `RadError` cannot be converted to `RadonError` because the error code is not defined
+    #[fail(
+        display = "`RadError` cannot be converted to `RadonError` because the error code is not defined"
+    )]
+    EncodeRadonErrorUnknownCode,
+    /// `RadError` cannot be converted to `RadonError` but it should, because it is needed for the tally result
+    #[fail(
+        display = "`RadError` cannot be converted to `RadonError` but it should, because it is needed for the tally result. Message: {:?}. Inner: `{:?}`",
+        message, inner
+    )]
+    UnhandledIntercept {
+        inner: Option<Box<RadError>>,
+        message: Option<String>,
+    },
+}
+
+impl RadError {
+    pub fn try_from_kind_and_cbor_args(
```

Comment:
As discussed, this is worth being separated from `RadError` and directly returning a `RadonError`. That way, the `match kind` would be much simpler, as it would only need to cover those `RadonErrors` for which we expect / support argument decoding.

Reply:
![image](https://user-images.githubusercontent.com/1316439/72905802-624b1100-3d31-11ea-9d71-1134e8dbcfad.png)